### ISSUE TITLE
Add the Extended Standard or Allow for custom directive

### ISSUE THREAD

**Issue by megamanhxh** [karl-sjogren/robots-txt-middleware#8]
It would be nice to have the extended standard for robot.txt within SectionBuilder.

The Extended Standard for Robot Exclusion was developed by Sean Conner, can be found here:
http://www.conman.org/people/spc/robots2.html

Or at least, allowing to add any custom directive in the SectionBuilder
```c#
builder.Services.AddStaticRobotsTxt(robotsTxtOptionsBuilder =>
    robotsTxtOptionsBuilder
        .AddSection(section =>
            section
                .AddComment("Allow crawlers to visit on these times")
                .AddUserAgent("*")
                .AddCustomDirective("Request-rate", "1/10")
                .AddCustomDirective("Visit-time", "0145-0830")
                ...
        )
        ...
);
```
Results
```txt
# Allow crawlers to visit on these times
User-agent: *
Request-rate: 1/10
Visit-time: 0145-0830
```

**Comment by karl-sjogren** [comment]
Adding a `CusomDirective` method is something I could be open to.

I can't find much information about the "extended standard" though. It seems to be quite old as well (last updated 20 years ago). Are there any known crawlers today that supports any of it?

**Comment by megamanhxh** [comment]
Honestly, I couldn't find any crawlers might use those extended standards. However, our SOC team has been forcing us to add them just-in-case a crawler might use them.

**Comment by karl-sjogren** [comment]
I'll look into adding a `CustomDirective` method then, it will be good enough for your scenario without bloating the main interface with what is probably an obsolete standard. It should be a fairly easy fix so I'll try to do it later this week.

**Comment by karl-sjogren** [comment]
I've released version 2.2.0 which adds a method called `AddCustomDirective(string directive, string value)`.

```
builder
    .AddSection(section =>
        section
            .AddComment("Allow Googlebot")
            .AddUserAgent("Googlebot")
            .Allow("/")
            .AddCustomDirective("Request-rate", "1/10")
        )
```

https://www.nuget.org/packages/RobotsTxtCore/2.2.0
